### PR TITLE
Tuning strategy components

### PR DESCRIPTION
**Pull request by gd1989**
I think what a tuning strategy entails could be stated explicitly

### CODE REVIEW

**Comment by tlienart** [comment]
@ablaom with the view that people reading tutorials might not want to read the docs, what would be an incomplete-but-useful short overview of what a tuning strategy is?

**Comment by ablaom** [comment]
Fair enough!! 

Does this help?

---

More precisely, "wrapping a model in a tuning strategy" means
creating a new "self-tuning" version of the model, `tuned_model =
TunedModel(model=...)`, in which further key-word arguments specify:

- the *algorithm* (a.k.a., *tuning strategy*) for searching the
  hyper-parameter space of the model (e.g., `tuning=Random(rng=123)`
  or `tuning=Grid(goal=100)`)
  
- the *resampling strategy*, used to evaluate performance for each
  value of the hyper-parameters (e.g., `resampling=CV(nfolds=9,
  rng=123)` or `resampling=Holdout(fraction_train=0.7)`)

- the *measure* (or measures) on which to base performance evaluations
  (and for reporting purposes) (e.g., `measure = rms` or `measures=[rms,
  mae]`)
  
- the *range*, usually describing the "space" of hyperparameters to be
  searched (but more generally whatever extra information is required
  to complete the search specification, e.g., initial values in
  gradient-descent optimization)
  
For more options do `?TunedModel`.

**Comment by gd1989** [comment]
@ablaom thanks for this. Just brought this in.

File: _literate/A-model-tuning.jl [hunk 1]
```diff
@@ -27,11 +27,18 @@ r   = range(dtc, :max_depth, lower=1, upper=5)
...
 # For hyperparameters of other type (e.g. `Symbol`), you should use the `values=...` keyword.
 #
-# Once a range of values has been defined, you can then wrap the model in a `TunedModel` specifying the tuning strategy. 
-# A tuning strategy consists of (i) a model, (ii) a set of values for the tuned paramters and (iii) a measure to optimize:
+# Once a range of values has been defined, you can then wrap the model in a `TunedModel` specifying the tuning strategy.
 
 tm = TunedModel(model=dtc, ranges=[r, ], measure=cross_entropy)
 
+# Note that "wrapping a model in a tuning strategy" as above means creating a new "self-tuning" version of the model, tuned_model = TunedModel(model=...), in which further key-word arguments specify:
+# 1. the algorithm (a.k.a., tuning strategy) for searching the hyper-parameter space of the model (e.g., tuning=Random(rng=123)or tuning=Grid(goal=100)).
```
Comment (tlienart) [inline _literate/A-model-tuning.jl]: please use backticks around code `` ` ``

Comment (gd1989) [reply _literate/A-model-tuning.jl]: I corrected it in second commit.

File: _literate/A-model-tuning.jl [hunk 2]
```diff
@@ -31,10 +31,10 @@ r   = range(dtc, :max_depth, lower=1, upper=5)
 
 tm = TunedModel(model=dtc, ranges=[r, ], measure=cross_entropy)
 
-# Note that "wrapping a model in a tuning strategy" as above means creating a new "self-tuning" version of the model, tuned_model = TunedModel(model=...), in which further key-word arguments specify:
-# 1. the algorithm (a.k.a., tuning strategy) for searching the hyper-parameter space of the model (e.g., tuning=Random(rng=123)or tuning=Grid(goal=100)).
-# 2. the resampling strategy, used to evaluate performance for each value of the hyper-parameters (e.g., resampling=CV(nfolds=9, rng=123) or resampling=Holdout(fraction_train=0.7)).
-# 3. the measure (or measures) on which to base performance evaluations (and for reporting purposes) (e.g., measure = rms or measures=[rms, mae]).
+# Note that "wrapping a model in a tuning strategy" as above means creating a new "self-tuning" version of the model, `tuned_model = TunedModel(model=...)`, in which further key-word arguments specify:
+# 1. the algorithm (a.k.a., tuning strategy) for searching the hyper-parameter space of the model (e.g., `tuning=Random(rng=123)or tuning=Grid(goal=100)`).
```
Comment (tlienart) [inline _literate/A-model-tuning.jl]: please use two backticks in between (so that the "or" is not in backticks) so 

```
`tuning = Random(rng=123)` or `tuning = Grid(goal=100)`
```

I recommend using hackmd.io to directly see what markdown looks like if you want.

Comment (ablaom) [reply _literate/A-model-tuning.jl]: This is still a tiny bit confusing (too many cooks, maybe). How about:

Once a range of values has been defined, you can then wrap the model in a `TunedModel`: ~~specifying the tuning strategy.~~

tm = TunedModel(model=dtc, ranges=[r, ], measure=cross_entropy)

~~Note that "wrapping a model in a tuning strategy" as above means~~ This essentially ~~creating~~ creates a new "self-tuning" version of the model, ...
